### DISCUSSION TITLE
Full Body Consistency

### DISCUSSION THREAD

**Discussion by Ambratolm** [lllyasviel/Fooocus#1020] [Ideas]
# Problematic

Is there a good or decent way to achieve full body consistency ? Not only face consistency.
I know it maybe hard or impossible for the moment, but any thoughts or ideas on the subject would be appreciated.

**Ultimate objectif:**
Given a character defined with an image or more of his full body (face, body shape, and cloths), incorporate him in a scene, either via a simple text prompt or via image prompt with a target scene image for body swapping.

For example, we want this guy here :
<details>
<summary>🖼 Source body image</summary>

![2023-11-24_04-10-15_6795](https://github.com/lllyasviel/Fooocus/assets/27871399/0da2577c-425c-42e0-a2fe-103e4c2cf39a)

</details>

to be fully (body, face, cloths)...

In this target image:

<details>
<summary>🖼 Target scene image</summary>

![2023-11-24_04-15-08_9748](https://github.com/lllyasviel/Fooocus/assets/27871399/0f70b5df-42b8-4825-9cb6-8c88ef9f814f)

</details>

or in an image generated by this prompt: 
```
strong 1man, jumping up, shooting cars, pointing guns on cops, epic city background with fire and destruction.
```

<hr />

# Attempt: Consistent Character Guide (Imperfect)

## 1. Character Reference Sheet Source

Find a good 3D character reference sheet source image online.

- 🔎 Some Search Links:

  - [_3d model reference sheet_](https://yandex.com/images/search?text=3d%20model%20reference%20sheet)
  - [_3d mannequin reference sheet_](https://yandex.com/images/search?text=3d%20mannequin%20reference%20sheet)
  - [_3d man reference sheet_](https://yandex.com/images/search?text=3d%20man%20reference%20sheet)
  - [_3d woman reference sheet_](https://yandex.com/images/search?text=3d%20woman%20reference%20sheet)

- ☑️ Example:

    <details>
    <summary>🖼 Character reference sheet source image</summary>

    ![ref0](https://github.com/lllyasviel/Fooocus/assets/27871399/f724be85-8031-4ad7-80bf-3219d47b20df)
    </details>

## 2. Character Reference Sheet

Generate your character sheet in Fooocus with a **text prompt** and an **image prompt**.

- Text Prompt:

  ```
  character turnaround, multiple views, [Body], [Style], [Cloths]
  ```

  - Replace `[Body]` with your description of the character's **body** and **face**.
  - Replace `[Style]` with your description of the additional **art style**.
  - Replace `[Cloths]` with your description of the character's **clothing**.

- Image Prompt:

  - Image: **Character Reference Sheet Source**
  - Type: `PyraCanny`
  - StopAt: `1`
  - Weight: `<= 0.5` (~ `0.35` Recommended)

- ☑️ Example:

  ```
  character turnaround, multiple views, red haired 1girl, 6 years old, short, blue eyes, Pixar styled character, black t-shirt, green pants, black shoes
  ```

  - **Body**: `red haired 1girl, 6 years old, short, blue eyes`
  - **Cloths**: `black t-shirt, green pants, black shoes`
  - **Style**: `pixar styled character`

  <details>
    <summary>🖼 Character reference sheet image</summary>

    ![ref1](https://github.com/lllyasviel/Fooocus/assets/27871399/d8721966-0a8c-4288-9d19-ba47286f5388)
  </details>

## 3. Character Reference Sheet without Background

Remove the background of your character reference sheet image and replace it with transparence, using any image editting tool that allows you to do it easily in one click, like [Adobe Express](https://new.express.adobe.com/tools/remove-background).

- Example:

  <details>
    <summary>🖼 Character reference sheet image wihout background</summary>

  ![ref2](https://github.com/lllyasviel/Fooocus/assets/27871399/48cb71b9-15ec-463b-a3b0-e82a75f103b6)
  </details>

## 4. Character Scene

Generate a scene with your character in Fooocus with a **text prompt** and **4 images prompt**.

- Text Prompt: 
  ```
  [Body], [Style], [Pose], [Cloths], [Background]
  ```
  - Replace `[Body]` with the same as before.
  - Replace `[Style]` with the same as before.
  - 🆕 Replace `[Pose]` with your description of the character's **physical pose** and **facial expression**.
  - Replace `[Cloths]` with the same as before.
  - 🆕 Replace `[Background]` with your description of the scene's **background**.

- Image Prompts:

  For each empty image slot (4 max):
  - Image: **Character Reference Sheet without Background**.
    - Use the `Edit` tool to **crop** a pose. Use a different pose in each slot.
  - Type: `ImagePrompt`
  - StopAt: `0.5` (Default)
  - Weight: `0.6` (Default)

- ☑️ Example:

  ```
  red haired 1girl, 6 years old, short, blue eyes, Pixar styled character, dynamic running pose, full sprint, laughing face, black t-shirt, green pants, black shoes, desert background
  ```

  <details>
      <summary>🖼 Character scene images</summary>

    ![scene (1)](https://github.com/lllyasviel/Fooocus/assets/27871399/bf5b39e1-cdc3-4b91-9292-b842b1f045ca)
    ![scene (2)](https://github.com/lllyasviel/Fooocus/assets/27871399/1e0ede1c-913b-458d-a45e-176b666aca13)
    ![scene (3)](https://github.com/lllyasviel/Fooocus/assets/27871399/217ffeeb-6155-409b-9ba5-a4c851b332d1)
    ![scene (4)](https://github.com/lllyasviel/Fooocus/assets/27871399/8607b919-2cbb-438b-9ecd-23f5d7beecad)
  </details>

**Comment by ayushyadav547** [comment]
i followed the whole process, yet i am getting a single image of the character
